### PR TITLE
Fix strict component name type safety

### PR DESCRIPTION
**Pull request by harijoe**
- Fixes the type safety provided by the new skybridge API
- We use type assertion in tests as the generated type ViewRegistry that we would expect normally is not available in tests
- Displays deepest typescript error instead of shallowest one

image

<h3>Greptile Summary</h3>

This PR makes two improvements: it tightens `ViewName` to `keyof ViewNameRegistry & string` (removing the `never`-fallback conditional type) for strict component-name type safety, and it overhauls `use-typescript-check.ts` to buffer multi-line TypeScript error output and surface the deepest (most specific) nested error message instead of the shallowest one. Test files are updated with `as ViewName` casts to satisfy the stricter type.

<h3>Confidence Score: 5/5</h3>

This PR is safe to merge; all deliberate trade-offs (never in tests, strict ViewName) are acknowledged by the team.

No P0 or P1 issues found. The buffering logic in use-typescript-check.ts is correct: flushPending is called at both error-header boundaries and the 'Found N errors' sentinel, continuation lines are properly scoped to the pending error, and extractBestMessage handles all edge cases. The as ViewName / as never cast in tests is intentional and confirmed by the team.

No files require special attention.

<sub>Reviews (3): Last reviewed commit: ["Fix lint"](https://github.com/alpic-ai/skybridge/commit/f3919827231504ad128daddafa109319993c4f55) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=30179352)</sub>

### CODE REVIEW

File: packages/core/src/server/server.ts [hunk 1]
```diff
@@ -85,9 +85,7 @@ export interface ViewCsp {
 // biome-ignore lint/suspicious/noEmptyInterface: register pattern — augmented by `.skybridge/views.d.ts` to narrow ViewName
 export interface ViewNameRegistry {}
 
-export type ViewName = keyof ViewNameRegistry extends never
-  ? string
-  : keyof ViewNameRegistry & string;
+export type ViewName = keyof ViewNameRegistry & string;
```
[the remark this answers is not in the document]
Comment (harijoe) [reply packages/core/src/server/server.ts:88]: This is expected

File: packages/core/src/test/utils.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export function createTestServer() {
           ),
           totalCount: z.number(),
         },
-        view: { component: "search-trip" },
+        view: { component: "search-trip" as ViewName },
```
[the remark this answers is not in the document]
Comment (harijoe) [reply packages/core/src/test/utils.ts:50]: We're okay to silent TS typechecker for the component names in tests

**Review by paulleseute — APPROVED** [review]
